src/machine_learning/api.py

```diff
@@ -112,5 +106,56 @@ def add_controlnet_segmentation(self, model: str, seg_map: str, depth_map: str =
             ],
         }
 
-    def to_dict(self) -> dict:
-        return self.config
+class Img2ImgConfig(X2ImgConfig):
```

Comment:
I would recommend splitting out classes into their own separate files to ensure that its easier to maintain and to avoid having extremely long files.